File: current_session/python/1822.py
"""
1822
"""
from typing import List
class Solution:
    def arraySign(self, nums: List[int]) -> int:
        cnt = 0
        for n in nums:
            if n == 0:
                return 0
            if n < 0:
                cnt += 1
        return -1 if cnt%2 else 1

File: current_session/c++/1822.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
  int arraySign(vector<int>& nums) {
    int negativeCnt = 0;
    for (int i=0; i<nums.size(); ++i) {
      if (nums[i] == 0) return 0;
      if (nums[i] < 0) negativeCnt++;
    }
    return (negativeCnt%2 ? -1 : 1);
  }
};

int main() {
  int x;
  cin >> x;     // length of array
  vector<int> arr;
  while (x>0) {
    int a;
    cin >> a;
    arr.push_back(a);
    --x;
  }
  cout << "result: " << Solution().arraySign(arr) << endl;

}